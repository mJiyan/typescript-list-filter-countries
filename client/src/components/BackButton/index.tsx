import React from 'react';
import { useHistory } from 'react-router-dom'


const BackButton = () => {
  const history = useHistory();

  return (
    <div
      className="
        xl:pt-10  lg:pt-10  md:pt-10  sm:pt-5  xs:pt-5  
    "
    >
      <button
        className="flex flex-row items-center rounded bg-WhiteDark dark:bg-DarkBlue text-sm shadow-lg relative
        xl:h-10  lg:h-10  md:h-10  sm:h-8  xs:h-8
        xl:px-10  lg:px-10  md:px-10  sm:px-4  xs:px-4
        xl:mt-0 lg:mt-0 md:mt-0 sm:mt-5 xs:mt-5
      "
        onClick={() =>history.goBack()}
        data-testid="back-button"
      >
        <svg
          xmlns="http://www.w3.org/2000/svg"
          className="h-5 w-5"
          viewBox="0 0 20 20"
          fill="currentColor"
        >
          <path
            fillRule="evenodd"
            d="M7.707 14.707a1 1 0 01-1.414 0l-4-4a1 1 0 010-1.414l4-4a1 1 0 011.414 1.414L5.414 9H17a1 1 0 110 2H5.414l2.293 2.293a1 1 0 010 1.414z"
            clipRule="evenodd"
          />
        </svg>
        <p className="px-2">Back</p>
      </button>
    </div>
  );
};

export default BackButton;
