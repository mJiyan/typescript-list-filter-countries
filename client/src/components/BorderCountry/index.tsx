import React from 'react';
import { Link } from 'react-router-dom'
import { IBorderCountry } from '../../models';

const BorderCountry: React.FC<IBorderCountry> = ({ country }) => {


  return (
    <div>
      <Link
        className="flex items-center justify-center rounded bg-WhiteDark dark:bg-DarkBlue text-sm shadow-lg truncate
          xl:text-sm lg:text-sm  md:text-sm  sm:text-sm  xs:text-sm
          xl:w-28  lg:w-28  md:w-28  sm:w-24  xs:w-24
          xl:h-8  lg:h-8  md:h-8  sm:h-8  xs:h-8"
        to={'/name/' + country.alpha}
      >
        {country.name}
      </Link>
    </div>
  );
};

export default BorderCountry;
