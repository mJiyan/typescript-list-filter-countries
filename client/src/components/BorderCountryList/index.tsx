import React from 'react';
import { IBorderCountryList } from '../../models';
import { BorderCountry } from '../';

const BorderCountryList: React.FC<IBorderCountryList> = ({ countries }) => {
  return (
    <div className="absolute">
      <div
        className="flex flex-col
          xl:flex-row lg:flex-row md:flex-row sm:flex-col xs:flex-col
          py-10 pr-5 w-full relative"
      >
        <p className="xl:pr-2 lg:pr-2 md:pr-2 sm:pr-0 xs:pr-0 pt-2">Border Countries:</p>
        <div className="grid 
            xl:grid-cols-4 lg:grid-cols-4 md:grid-cols-4 sm:grid-cols-3 xs:grid-cols-3
            xl:pt-0  lg:pt-0  md:pt-0   sm:pt-2   xs:pt-2"
        >
          {countries &&
            countries.map((item, key) => (
              <div className="pr-1 py-1" key={key}>
                <BorderCountry key={key} country={item} />
              </div>
            ))}
        </div>
      </div>
    </div>
  );
};

export default BorderCountryList;
