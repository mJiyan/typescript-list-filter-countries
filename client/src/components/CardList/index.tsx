import React from 'react';
import { ICardList } from '../../models';
import Card from '../Card';

const CardList: React.FC<ICardList> = ({ cards }) => {
  const array = [0, 1, 2, 3, 4, 5];
  return (
    <div
      className="
        grid xl:grid-cols-4 lg:grid-cols-4  md:grid-cols-2  sm:grid-cols-1  xs:grid-cols-1 justify-items-center 
        xl:pt-10 lg:pt-10 md:pt-10 sm:pt-0 xs:pt-0 px-10 
    "
    >
      {cards &&
        cards.map((item, key) => (
          <Card
            key={item.name}
            name={item.name}
            population={item.population}
            region={item.region}
            capital={item.capital}
            flag={item.flag}
            alpha3Code={item.alpha3Code}
          />
        ))}
    </div>
  );
};

export default CardList;
