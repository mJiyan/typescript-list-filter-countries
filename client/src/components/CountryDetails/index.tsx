import React from 'react';
import { ICountryDetails } from '../../models';
import { CountryInformations, Flag, BackButton } from '../';

const CountryDetails: React.FC<ICountryDetails> = ({
  name,
  flag,
  nativeName,
  population,
  region,
  subRegion,
  capital,
  topLevelDomain,
  currencies,
  languages,
  borderCountries,
}) => {
  return (
    <div className="flex flex-col w-full h-screen 
      xl:px-20 lg:px-20 md:px-20 sm:px-5 xs:px-3
    ">
      <div className="absolute">
        <BackButton />
      </div>
      <div
        className="flex 
          xl:flex-row lg:flex-row md:flex-row sm:flex-col xs:flex-col 
          xl:pt-36 lg:pt-36 md:pt-36  sm:pt-0  xs:pt-0 
          h-screen "
      >
        <Flag flag={flag} />
        <CountryInformations
          name={name}
          nativeName={nativeName}
          population={population}
          region={region}
          subRegion={subRegion}
          capital={capital}
          topLevelDomain={topLevelDomain}
          currencies={currencies}
          languages={languages}
          borderCountries={borderCountries}
        />
      </div>
    </div>
  );
};

export default CountryDetails;
