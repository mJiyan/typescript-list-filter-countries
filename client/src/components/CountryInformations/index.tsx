import React from 'react';
import { BorderCountryList } from '../';
import { ICountryInformations } from '../../models';
import { numberWithCommas } from '../../services/utils'

const CountryInformations: React.FC<ICountryInformations> = ({
  name,
  nativeName,
  population,
  region,
  subRegion,
  capital,
  topLevelDomain,
  currencies,
  languages,
  borderCountries,
}) => {
  return (
    <div
      className="
      xl:w-1/2 lg:w-1/2 md:w-1/2 sm:w-full xs:w-full
      xl:mt-0 lg:mt-0 md:mt-0 sm:mt-12 xs:mt-12
    "
    >
      <p className="text-4xl pb-8 w-full font-bold">{name}</p>
      <div
        className="flex
          xl:flex-row lg:flex-row md:flex-row sm:flex-col xs:flex-col 
          xl:h-fit lg:h-fit md:h-fit sm:h-full xs:h-full
          items-between justify-between w-full "
      >
        <div className="">
          <p>
            <span className='font-normal'>Native Name:</span> <span className='font-light'>{nativeName}</span>
          </p>
          <p>
            <span className='font-normal'>Population:</span> <span className='font-light'>{numberWithCommas(parseInt(population))}</span>
          </p>
          <p>
            <span className='font-normal'>Region:</span> <span className='font-light'>{region}</span>
          </p>
          <p>
            <span className='font-normal'>Sub Region:</span> <span className='font-light'>{subRegion}</span>
          </p>
          <p>
            <span className='font-normal'>Capital:</span> <span className='font-light'>{capital}</span>
          </p>
        </div>
        <div className="">
          <p>
            <span className='font-normal'>Top Level Domain:</span> <span className='font-light'>{topLevelDomain}</span>
          </p>
          <p>
            <span className='font-normal'>Currencies:</span> <span className='font-light'>{currencies}</span>

          </p>
          <p>
            <span className='font-normal'>Languages: {' '}</span> <span className='font-light'>{languages.map((item, key) => {
              return item.name + (languages.length !== key + 1 ? ', ' : '');
            })}</span>

          </p>
        </div>
      </div>
      <div>
        <BorderCountryList countries={borderCountries} />
      </div>
    </div>
  );
};

export default CountryInformations;
