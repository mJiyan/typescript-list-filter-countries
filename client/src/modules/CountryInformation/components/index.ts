export { default as CountryInformationText } from "./CountryInformationText";
export { default as BorderCountryList } from "./BorderCountryList";
