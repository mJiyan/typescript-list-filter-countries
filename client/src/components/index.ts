import Header from './Header';
import Filter from './Filter';
import Card from './Card';
import CardList from './CardList';
import CountryInformations from './CountryInformations';
import CountryDetails from './CountryDetails';
import Flag from './Flag'
import BorderCountry from './BorderCountry'
import BorderCountryList from './BorderCountryList'
import BackButton from './BackButton'

export { Header, Filter, Card, CardList, CountryDetails, CountryInformations, Flag, BorderCountry, BorderCountryList, BackButton };
