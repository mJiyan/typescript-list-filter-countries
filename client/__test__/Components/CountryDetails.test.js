import React from 'react';
import { CountryDetails } from '../../src/components';
import renderer from 'react-test-renderer';

import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import { Router, Route, Switch } from 'react-router-dom';
import { createMemoryHistory } from 'history'

import defaultState from '../Mock/defaulState';
import toJson from 'enzyme-to-json'; //added this line
import 'jsdom-global/register'; //at the top of file , even  , before importing react

const mockStore = configureMockStore([thunk]);
const store = mockStore(defaultState);
const history = createMemoryHistory()

const wrapper = renderer.create(
  <Provider store={store}>
    <Router history={history}>
      <Switch>
        <CountryDetails
          name={defaultState[0].name}
          flag={defaultState[0].flag}
          nativeName={defaultState[0].nativeName}
          population={defaultState[0].population}
          region={defaultState[0].region}
          subRegion={defaultState[0].subregion}
          capital={defaultState[0].capital}
          topLevelDomain={defaultState[0].topLevelDomain}
          currencies={defaultState[0].currencies[0].name}
          languages={defaultState[0].languages}
          borderCountries={defaultState[0].borders}
        />
      </Switch>

    </Router>

  </Provider>,
);


describe('rendering countryDetails components', () => {
  test('it should match with the snapshot', () => {
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

});
