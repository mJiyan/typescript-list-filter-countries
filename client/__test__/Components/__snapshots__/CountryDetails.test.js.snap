// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering countryDetails components it should match with the snapshot 1`] = `
<div
  className="flex flex-col w-full h-screen 
      xl:px-20 lg:px-20 md:px-20 sm:px-5 xs:px-3
    "
>
  <div
    className="absolute"
  >
    <div
      className="
        xl:pt-10  lg:pt-10  md:pt-10  sm:pt-5  xs:pt-5  
    "
    >
      <button
        className="flex flex-row items-center rounded bg-WhiteDark dark:bg-DarkBlue text-sm shadow-lg relative
        xl:h-10  lg:h-10  md:h-10  sm:h-8  xs:h-8
        xl:px-10  lg:px-10  md:px-10  sm:px-4  xs:px-4
        xl:mt-0 lg:mt-0 md:mt-0 sm:mt-5 xs:mt-5
      "
        data-testid="back-button"
        onClick={[Function]}
      >
        <svg
          className="h-5 w-5"
          fill="currentColor"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M7.707 14.707a1 1 0 01-1.414 0l-4-4a1 1 0 010-1.414l4-4a1 1 0 011.414 1.414L5.414 9H17a1 1 0 110 2H5.414l2.293 2.293a1 1 0 010 1.414z"
            fillRule="evenodd"
          />
        </svg>
        <p
          className="px-2"
        >
          Back
        </p>
      </button>
    </div>
  </div>
  <div
    className="flex 
          xl:flex-row lg:flex-row md:flex-row sm:flex-col xs:flex-col 
          xl:pt-36 lg:pt-36 md:pt-36  sm:pt-0  xs:pt-0 
          h-screen "
  >
    <div
      className="flex flex-col justify-end items-start
        xl:h-1/2 lg:h-1/2 md:h-1/2 sm:h-1/2 xs:h-1/2
        xl:w-1/2 lg:w-1/2 md:w-1/2 sm:w-full xs:w-full
        xl:mt-0 lg:mt-0 md:mt-0 sm:mt-12 xs:mt-12
    "
    >
      <div
        className="
        xl:w-5/6 lg:w-5/6 md:w-5/6 sm:w-full xs:w-full
        xl:h-full lg:h-full md:h-full sm:h-3/4 xs:h-3/4
        bg-center bg-cover  bg-no-repeat"
        style={
          Object {
            "backgroundImage": "url(https://flagcdn.com/be.svg)",
          }
        }
      />
    </div>
    <div
      className="
      xl:w-1/2 lg:w-1/2 md:w-1/2 sm:w-full xs:w-full
      xl:mt-0 lg:mt-0 md:mt-0 sm:mt-12 xs:mt-12
    "
    >
      <p
        className="text-4xl pb-8 w-full font-bold"
      >
        Belgium
      </p>
      <div
        className="flex
          xl:flex-row lg:flex-row md:flex-row sm:flex-col xs:flex-col 
          xl:h-fit lg:h-fit md:h-fit sm:h-full xs:h-full
          items-between justify-between w-full "
      >
        <div
          className=""
        >
          <p>
            <span
              className="font-normal"
            >
              Native Name:
            </span>
             
            <span
              className="font-light"
            >
              BelgiÃ«
            </span>
          </p>
          <p>
            <span
              className="font-normal"
            >
              Population:
            </span>
             
            <span
              className="font-light"
            >
              11,555,997
            </span>
          </p>
          <p>
            <span
              className="font-normal"
            >
              Region:
            </span>
             
            <span
              className="font-light"
            >
              Europe
            </span>
          </p>
          <p>
            <span
              className="font-normal"
            >
              Sub Region:
            </span>
             
            <span
              className="font-light"
            >
              Western Europe
            </span>
          </p>
          <p>
            <span
              className="font-normal"
            >
              Capital:
            </span>
             
            <span
              className="font-light"
            >
              Brussels
            </span>
          </p>
        </div>
        <div
          className=""
        >
          <p>
            <span
              className="font-normal"
            >
              Top Level Domain:
            </span>
             
            <span
              className="font-light"
            >
              .be
            </span>
          </p>
          <p>
            <span
              className="font-normal"
            >
              Currencies:
            </span>
             
            <span
              className="font-light"
            >
              Euro
            </span>
          </p>
          <p>
            <span
              className="font-normal"
            >
              Languages: 
               
            </span>
             
            <span
              className="font-light"
            >
              Dutch, 
              French, 
              German
            </span>
          </p>
        </div>
      </div>
      <div>
        <div
          className="absolute"
        >
          <div
            className="flex flex-col
          xl:flex-row lg:flex-row md:flex-row sm:flex-col xs:flex-col
          py-10 pr-5 w-full relative"
          >
            <p
              className="xl:pr-2 lg:pr-2 md:pr-2 sm:pr-0 xs:pr-0 pt-2"
            >
              Border Countries:
            </p>
            <div
              className="grid 
            xl:grid-cols-4 lg:grid-cols-4 md:grid-cols-4 sm:grid-cols-3 xs:grid-cols-3
            xl:pt-0  lg:pt-0  md:pt-0   sm:pt-2   xs:pt-2"
            >
              <div
                className="pr-1 py-1"
              >
                <div>
                  <a
                    className="flex items-center justify-center rounded bg-WhiteDark dark:bg-DarkBlue text-sm shadow-lg truncate
          xl:text-sm lg:text-sm  md:text-sm  sm:text-sm  xs:text-sm
          xl:w-28  lg:w-28  md:w-28  sm:w-24  xs:w-24
          xl:h-8  lg:h-8  md:h-8  sm:h-8  xs:h-8"
                    href="/name/FRA"
                    onClick={[Function]}
                  >
                    France
                  </a>
                </div>
              </div>
              <div
                className="pr-1 py-1"
              >
                <div>
                  <a
                    className="flex items-center justify-center rounded bg-WhiteDark dark:bg-DarkBlue text-sm shadow-lg truncate
          xl:text-sm lg:text-sm  md:text-sm  sm:text-sm  xs:text-sm
          xl:w-28  lg:w-28  md:w-28  sm:w-24  xs:w-24
          xl:h-8  lg:h-8  md:h-8  sm:h-8  xs:h-8"
                    href="/name/DEU"
                    onClick={[Function]}
                  >
                    Germany
                  </a>
                </div>
              </div>
              <div
                className="pr-1 py-1"
              >
                <div>
                  <a
                    className="flex items-center justify-center rounded bg-WhiteDark dark:bg-DarkBlue text-sm shadow-lg truncate
          xl:text-sm lg:text-sm  md:text-sm  sm:text-sm  xs:text-sm
          xl:w-28  lg:w-28  md:w-28  sm:w-24  xs:w-24
          xl:h-8  lg:h-8  md:h-8  sm:h-8  xs:h-8"
                    href="/name/LUX"
                    onClick={[Function]}
                  >
                    Luxembourg
                  </a>
                </div>
              </div>
              <div
                className="pr-1 py-1"
              >
                <div>
                  <a
                    className="flex items-center justify-center rounded bg-WhiteDark dark:bg-DarkBlue text-sm shadow-lg truncate
          xl:text-sm lg:text-sm  md:text-sm  sm:text-sm  xs:text-sm
          xl:w-28  lg:w-28  md:w-28  sm:w-24  xs:w-24
          xl:h-8  lg:h-8  md:h-8  sm:h-8  xs:h-8"
                    href="/name/NLD"
                    onClick={[Function]}
                  >
                    Netherlands
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
